frontend/device/input.lua

```diff
@@ -459,20 +463,32 @@ function Input:handleTouchEvPhoenix(ev)
 end
 
 function Input:handleOrientationEv(ev)
-
     if ev.type == EV_ABS then
-        if ev.code == ABS_PRESSURE and self.device.screen:getScreenMode() == 'portrait' then
-            local refreshUI = false
+        if ev.code == ABS_PRESSURE then
+
             local rotation_mode = 0
-            if ev.value == DEVICE_ORIENTATION_PORTRAIT then
-                refreshUI = true
-            elseif ev.value == DEVICE_ORIENTATION_PORTRAIT_ROTATED then
-                refreshUI = true
+            local screen_mode = 'portrait'
+            if ev.value == DEVICE_ORIENTATION_PORTRAIT or ev.value == DEVICE_ORIENTATION_PORTRAIT_LEFT or ev.value == DEVICE_ORIENTATION_PORTRAIT_RIGHT then
+                rotation_mode = 0
+                screen_mode = 'portrait'
+            elseif ev.value == DEVICE_ORIENTATION_LANDSCAPE then
+                rotation_mode = 1
+                screen_mode = 'landscape'
+            elseif ev.value == DEVICE_ORIENTATION_PORTRAIT_ROTATED or ev.value == DEVICE_ORIENTATION_PORTRAIT_ROTATED_LEFT or ev.value == DEVICE_ORIENTATION_PORTRAIT_ROTATED_RIGHT then
                 rotation_mode = 2
+                screen_mode = 'portrait'
+            elseif ev.value == DEVICE_ORIENTATION_LANDSCAPE_ROTATED then
+                rotation_mode = 3
+                screen_mode = 'landscape'
             end
 
-            local oldRotation = self.device.screen:getRotationMode()
-            if refreshUI and rotation_mode ~= oldRotation then
+            local old_rotation_mode = self.device.screen:getRotationMode()
+            local old_screen_mode = self.device.screen:getScreenMode();
+            DEBUG("old_rotation_mode: " .. old_rotation_mode)
```

Comment:
I recommend rewriting these debug print as:

```
DEBUG("old_rotation_mode:", old_rotation_mode)
```

Otherwise, the string concatenation operator will create new strings even when debug mode is disabled.